extensions/functions_arithmetic_decimal.yaml

```diff
@@ -102,14 +102,96 @@ scalar_functions:
           scale_after_borrow = max(init_scale - delta, min_scale)
           scale = init_prec > 38 ? scale_after_borrow : init_scale
           DECIMAL<prec, scale>
-  - 
+  -
     name: "abs"
     description: Calculate the absolute value of the argument.
     impls:
       - args:
           - name: x
             value: decimal<P,S>
         return: decimal<P,S>
+  -
+    name: "power"
+    description: "Take the power with x as the base and y as exponent."
+    impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P,S>"
+          - name: y
+            value: "DECIMAL<P,S>"
+        options:
+          overflow:
+            values: [ SILENT, SATURATE, ERROR ]
+        return: fp64
+  -
+     name: "sqrt"
+     description: "Square root of the value"
+     impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P,S>"
+        options:
+          rounding:
+            values: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+          on_domain_error:
+            values: [ NAN, ERROR ]
+        return: fp64
+  -
+     name: "factorial"
+     description: >
+      Return the factorial of a given decimal input.
+      The factorial of 0! is 1 by convention.
+      Negative inputs will raise an error.
+     impls:
+      - args:
+          - value: "DECIMAL<P,S>"
+            name: "n"
+        options:
+          overflow:
+            values: [ SILENT, SATURATE, ERROR ]
+        return: "DECIMAL<38,S>"
+  -
+     name: "bitwise_not"
+     description: >
+        Return the bitwise NOT result for one decimal input.
+     impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P,S>"
+        return: "DECIMAL<P,S>"
+  -
+     name: "bitwise_and"
+     description: >
+        Return the bitwise AND result for two decimal inputs.
+     impls:
+      - args:
+          - name: x
+            value: "DECIMAL<P,S>"
+          - name: y
+            value: "DECIMAL<P,S>"
+        return: "DECIMAL<P,S>"
```

Comment:
Isn't this some pattern of max of others?

Reply:
I have add this function in separate PR. I will take care of this comment in that future PR

Reply:
resolving since added this functions to #675